_posts/2023-10-24-filesystem-integrity-elytron-tool.adoc

```diff
@@ -0,0 +1,177 @@
+ODXvpN01Dgp74KjrbIvh9411HAndwEY/K9JVytrJYjgFm1wOS5q7CSRqeB9d2++CbVUxVFf2VrLD&#13;
+eoFGkN4hGyElPpPeL9yELTp3eZ71R/99H5jFLf21JHll/zqV3b1qreukdyuUUblBQ39/cYNMZxfX&#13;
+X813x7RDKE57HfkTdf/2Jv/se01vnMvvB36SL3UJ6GBcOz2IpCUyZ7CKGUptXBbUllSplYB4SyHv&#13;
+m9OOWBLuCIDbifdNrMmnxkVZRg+CPQy9c98bhA==</SignatureValue><KeyInfo><KeyValue><RSAKeyValue><Modulus>jj2pFiphnnkRssxcuinzz+qMlY9wgbMo7T0fXh/I2SQKOlbBjC1YJk0RJMCcmE1ITZbYk7JS5aiM&#13;
+fsY0qe6KNVLZLZcmp0vCruCTaIkNxKExfszezMxPN/NpyuxOPu+rhyX8Q4TkdYkIZZECuXUrocqf&#13;
+JiZHimuA7tmYaLOhJgN8ZspHITeKkp3hSiZIr61lz8DPWiR1P4+/LhzE4J/URphTOzh/y8tY6ySU&#13;
+3DPMDPjRHXUfkWbttd0HmzB+m5g8ZkT/9bB54HaP9FQHy6ynrywVKBS6+sm6RX/cTPeh1yoBkuyv&#13;
+IDHhPpbWb9nHrfDV10EF5N/MHBZEpThYXVIiyQ==</Modulus><Exponent>AQAB</Exponent></RSAKeyValue></KeyValue></KeyInfo></Signature></identity>
+```
+
+You can verify the integrity of the realm using the verify-integrity function: 
+```
+/subsystem=elytron/filesystem-realm=filesystem-realm-with-integrity:verify-integrity()
+```
+
+== Bulk Conversion for Filesystem Realms
+
+The Elytron tool also supports converting multiple filesystem realms at once. This can be done using the following command: 
+```
+$ WILDFLY_HOME/bin/elytron-tool.sh filesystem-realm-integrity --bulk-convert <descriptor> --summary
+```
+where `<descriptor>` refers to a descriptor file with details about the filesystems to be converted. 
+Here is an example descriptor file that converts 2 filesystem realms, `unsigned-realm-one` and `unsigned-realm-two` to add integrity checking. Note that we have specified the name of the new realm using the `realm-name` option to avoid getting assigned the default name. 
+```
+input-location:standalone/configuration/unsigned-realm-one
+output-location:standalone/configuration/signed
+keystore:myKeyStore
+password:myPassword
+key-pair:myKeyPair
+realm-name:realm-one
+
+input-location:standalone/configuration/unsigned-realm-two
+output-location:standalone/configuration/signed
+keystore:myKeyStore
+password:myPassword
+key-pair:myKeyPair
+realm-name:realm-two
+```
+
+== Summary
+
+This blog post has shown how to add integrity checking to an existing filesystem realm using an asymmetrical key pair and the WildFly Elytron tool. We also looked at how we can convert multiple filesystem realms using the `bulk-convert` option. Please note that in addition to elytron-tool.sh, WildFly also ships with `elytron-tool.bat` and `elytron-tool.ps1` scripts. For more information on the different options available for the elytron-tool, you can use the `--help` option. To learn more about the filesystem-security-realm, please refer to the https://docs.wildfly.org/30/WildFly_Elytron_Security.html#filesystem-security-realm[WildFly documentation]. 
```

Comment:
Just a minor, s/filesystem-security-realm/filesystem-realm or s/filesystem-security-realm/filesystem security realm

Reply:
Fixed! 
Thank you!